test/runner/runner.go

```diff
@@ -26,22 +26,58 @@ import (
 func NewRegistry() *TestRegistry {
 	return &TestRegistry{
 		tests: make(map[string]Test),
+		testGroups: make(map[string]TestGroup),
 	}
 }
 
-// TestRegistry contains a mapping of named tests
+//NewTestGroup returns a pointer to a new TestGroup
+func NewTestGroup(name string) *TestGroup {
+	return &TestGroup{
+		name: name,
+		tests: make(map[string]Test),
+	}
+}
+
+// Test is a test function
+type Test func(t *testing.T)
+
+
+// TestRegistry contains a mapping of named test groups
 type TestRegistry struct {
 	suite.SetupTestSuite
 	tests map[string]Test
+	testGroups map[string]TestGroup
 }
 
-// Register registers a named test
+//TestGroup to run multiple tests
+type TestGroup struct {
+	name string
+	suite.SetupTestSuite
+	tests map[string]Test
+}
+
+// AddToGroup registers a named test
+func (r *TestRegistry) AddToGroup(testName string, testGroup *TestGroup, test Test) {
```

Comment:
Also, there's a bit of an inconsistency with naming here. We should either use `Add` or `Register`, not both. 
It also may not make sense for `AddToGroup` to take a test name here when the test is already providing a name for `Register`. We need an API that ensures at least that:
* Tests that are added to a group are also added to the general registry so they can be run by name
* Tests cannot be added to the global registry and the group registry with different names

Reply:
Nevermind... I see there _is_ a `Register` receiver on `TestGroup`. We should just remove the `AddToGroup` version as it's redundant.